dags/sebra_dag.py

```diff
@@ -0,0 +1,92 @@
+import pendulum
+
+from airflow.decorators import dag, task
+
+import os
+import sys
+
+sebra_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
+sys.path.append(sebra_path)
+
+from src.sebra_scrape.constants import (
+    SEBRA_ID_STRINGS,
+    SEBRA_FILE_DESTINATIONS,
+    SEBRA_REPORT_ID_STRING,
+)
+
+
+# TODO: read from env?
+PATH_TO_SEBRA_PYTHON_BINARY = os.path.join(
```

Comment:
> At the moment in order to run the DAG the venv should be created on the airflow machine in the corresponding path, is that correct?

Correct. We pass the path to the python executable in that venv to the `ExternalPythonOperator` and Airflow arranges that operator to be executed in the existing venv. A lightweight alternative is a [`PythonVirtualenvOperator`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/python/index.html#airflow.operators.python.PythonVirtualenvOperator) where we say what packages we want in the virtualenv which is to run the operator and that venv is created on the fly each time the operator executes. The heavier alternatives are Docker and Kubernetes operators but those require setting up more infrastructure and maybe in the spirit of making things work before making them perfect, we can ignore them for now.

> Is that manual process at the moment? could it be automated? Are there any known good practices?
> Could another DAG be responsible to create/update the venv for all

I was thinking having a shell script that pulls changes from the remote and sets up the environment. If we set up all projects like this, we can have a DAG which loops over all projects and does that, yes. I think one more "standard" solution would be (`airflow-git-sync`)[https://github.com/data-burst/airflow-git-sync?tab=readme-ov-file#airflow-git-sync] (In my last team, we used the k8s version (`git-sync`)).  This would require setting up Docker and, I believe, having all of our DAGs live in a separate repo just for DAGs (that's the model I've worked with before, too)

Reply:
Writing the above, I think setting up our Airflow environment and development process might be a bigger, more general task. I am now thinking of keeping the current PR to just making the SEBRA DAG run in the most straightforward way possible that will not mess up our Airflow environment, which would be a `PythonVirtualenvOperator` and then we make a separate initiative to make a small design for the development process around DAGs with any interested parties. What do you think?

Reply:
Thanks for the info, I have food for thoughts :)
And yeah, the PR is good as it is, no need to do anything about venvs now, i was just curios.